lib/salus/scanners/report_go_dep.rb

```diff
@@ -1,47 +1,119 @@
 require 'toml'
 require 'salus/scanners/base'
+require 'salus/plugin_manager'
+require 'salus/report'
 
-# Report the use of Go packages captured in a Gopkg.lock files.
+# Report the use of Go packages captured in one of Gopkg.lock/go.sum/go.mod files.
 # https://github.com/golang/dep
 
 module Salus::Scanners
   class ReportGoDep < Base
     def run
       unless should_run?
-        raise(
-          InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+      end
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.foreach(go_sum_path).each("=\n") do |line|
+        line = line.strip
+        next if line.empty?
+
+        go_sum_regex = %r{(?<namespace>(.*)(?!/go\.mod))/(?<name>[^\s]*)
+        (\s)*(?<version>(.*))(\s)*h1:(?<checksum>(.*))}x
+
+        if (matches = line.match(go_sum_regex))
+          dep_list.append(
+            {
+              "namespace" => (matches[:namespace]).to_s,
+              "name" => (matches[:name]).to_s,
+              "version" => (matches[:version]).to_s.gsub(%r{/go.mod}, '').strip,
+              "checksum" => (matches[:checksum]).to_s
+            }
+          )
+        end
+      end
+      # Note references are hashes meant for packages in Gopkg.lock files
+      dep_list.each do |dependency|
+        record_dep_package(
+          namespace: dependency["namespace"],
+          name: dependency["name"],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.sum",
+          checksum: dependency['checksum'],
+          type: "golang"
         )
       end
+    end
+
+    def record_dep_from_go_mod
+      # Typically go.mod files don't contain all of the transitive deps/info
+      # a go.sum file would. Instead of parsing go.mod files, warn that
+      # go.sum is missing and send an event
+      warning_string = "WARNING: No go.sum/Gopkg.lock found. "\
+      "Currently go.mod is unsupported for reporting Golang dependencies."
+      report_warn(:report_go_dep_missing_go_sum, warning_string)
+
+      data = "This repository contains no go.sum or Gopkg.lock file. Currently "\
+      "go.mod files are unsupported for reporting Golang dependencies."
+      Salus::PluginManager.send_event(:report_go_dep_missing_go_sum, data)
```

Comment:
You can abstract the send event here to begin sending events for all report_warns.  That'll let you drop lines 67-69 here and provide a general report warn support for plugins.  Just be sure to add an entry for this event in https://github.com/coinbase/salus/blob/03279595de981a0aaad278f90f51d797e728e348/docs/architecture.md

`Salus::PluginManager.send_event(:report_warn, {type: type, message: message)
`
```
# Report a scanner warning such as a possible misconfiguration
    def report_warn(type, message)
      @report.warn(type, message)
      Salus::PluginManager.send_event(:report_warn, {type: type, message: message)
      if @salus_report&.builds
        scanner = @report.scanner_name
        message = "#{scanner} warning: #{type}, #{message}, build: #{@salus_report.builds}"
      end
      bugsnag_notify(message)
    end
```